budgetportal/models.py

```diff
@@ -768,8 +769,324 @@ def get_expenditure_over_time(self):
+            return None
+        phase_ref = openspending_api.get_phase_ref()
+        descript_ref = openspending_api.get_adjustment_kind_ref()
+        total_adjusted, total_voted = None, None
+
+        for cell in cells:
+            if cell[phase_ref] == 'Adjusted appropriation' and \
+                    cell[descript_ref] == 'Adjustments - Total adjustments':
+                total_adjusted = cell['value.sum']
+            if cell[phase_ref] == 'Voted (Main appropriation)' and \
+                    cell[descript_ref] == 'Total':
+                total_voted = cell['value.sum']
+
+        if total_voted and not total_adjusted:
+            total_adjusted = 0
+        elif not (total_voted or total_adjusted):
+            raise Exception("Could not calculate total change for department budget")
+
+        return total_voted, total_adjusted
+
+    def _get_budget_virements(self, openspending_api, dataset, total_voted):
+        virements_resource = dataset.get_resource('CSV', name='Value of Virements')
+        if virements_resource:
+            sql = '''
+                    SELECT "Value of Virements" FROM "{}"
+                    WHERE "department_name"='{}'
+                    '''.format(virements_resource['id'], self.name)
+            params = {
+                'sql': sql
+            }
+            result = requests.get(CKAN_DATASTORE_URL, params=params)
+            result.raise_for_status()
+            records = result.json()['result']['records']
+            if records:
+                value = records[0]['Value of Virements']
+            else:
+                raise Exception('Value of Virements query returned no records')
+            virements = {
+                'label': 'Value of virements',
+                'amount': int(value),
+                'percentage': 100 * (float(value) / float(total_voted))
+            }
+        else:
+            result_for_virements = openspending_api.aggregate(
+                cuts=[
+                    openspending_api.get_financial_year_ref() + ':' + self.get_financial_year().get_starting_year(),
+                    openspending_api.get_adjustment_kind_ref() + ':' +
+                    '"Adjustments - Virements and shifts due to savings"',
+
+                ],
+                drilldowns=openspending_api.get_all_drilldowns())
+
+            result_for_virements = openspending_api.filter_dept(result_for_virements, self.name)
+            cells_for_virements = openspending_api.filter_by_ref_exclusion(result_for_virements['cells'],
+                                                                           openspending_api.get_programme_name_ref(),
+                                                                           'Direct charge against the National '
+                                                                           'Revenue Fund')
+            total_positive_virement_change = 0
+            for c in cells_for_virements:
+                value = c['value.sum']
```

Comment:
it's cool to do assignment like this for readability. Don't worry about doing it for efficiency - you won't see the benefit of factoring out a dict lookup except in very special cases.

Reply:
It was definitely for readability, this is a programming habit of mine. Though I think in this case it's unnecessary as it's only used for one conditional and one operation